From bd188e306a06ca0acb6c213ecc0e1dc0dc22c3f6 Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Mon, 17 Apr 2023 16:23:06 +0100
Subject: [PATCH 31/41] Fix paren blunder in
 aaba66efbd3b4e7283993ca3718df47706a8549b

Thanks to Dominik Derigs for spotting this.
---
 src/tftp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tftp.c b/src/tftp.c
index caf077f..f036297 100644
--- a/src/tftp.c
+++ b/src/tftp.c
@@ -228,7 +228,7 @@ void tftp_request(struct listener *listen, time_t now)
 #ifdef HAVE_DHCP      
 	  /* allowed interfaces are the same as for DHCP */
 	  for (tmp = daemon->dhcp_except; tmp; tmp = tmp->next)
-	    if (tmp->name && (tmp->flags & (INAME_4 | INAME_6) == (INAME_4 | INAME_6)) &&
+	    if (tmp->name && (tmp->flags & INAME_4) && (tmp->flags & INAME_6) &&
 		wildcard_match(tmp->name, name))
 	      return;
 #endif
-- 
2.41.0

